api/filters/brokers_filter_by_visibility.go

```diff
@@ -0,0 +1,125 @@
+package filters
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/Peripli/service-manager/pkg/query"
+
+	"github.com/Peripli/service-manager/pkg/types"
+
+	"github.com/Peripli/service-manager/storage"
+
+	"github.com/Peripli/service-manager/pkg/web"
+)
+
+const BrokersVisibilityFilterName = "BrokersFilterByVisibility"
+
+func NewBrokersFilterByVisibility(repository storage.Repository) *BrokersFilterByVisibility {
+	return &BrokersFilterByVisibility{
+		visibilityFilteringMiddleware: &visibilityFilteringMiddleware{
+			FilteringFunc: newBrokersFilterFunc(repository),
+		},
+	}
+}
+
+type BrokersFilterByVisibility struct {
+	*visibilityFilteringMiddleware
+}
+
+func newBrokersFilterFunc(repository storage.Repository) func(ctx context.Context, platformID string) (*query.Criterion, error) {
+	return func(ctx context.Context, platformID string) (*query.Criterion, error) {
+		planQuery, err := plansCriteria(ctx, repository, platformID)
+		if err != nil {
+			return nil, err
+		}
+		if planQuery == nil {
+			return nil, nil
+		}
+
+		var servicesQuery *query.Criterion
+		servicesQuery, err = servicesCriteria(ctx, repository, planQuery)
+		if err != nil {
+			return nil, err
+		}
+		if servicesQuery == nil {
+			return nil, nil
+		}
+
+		brokersQuery, err := brokersCriteria(ctx, repository, servicesQuery)
+		if err != nil {
+			return nil, err
+		}
+		if brokersQuery == nil {
+			return nil, nil
+		}
+
+		return brokersQuery, nil
+	}
+}
+
+func brokersCriteria(ctx context.Context, repository storage.Repository, serviceQuery *query.Criterion) (*query.Criterion, error) {
+	objectList, err := repository.List(ctx, types.ServiceOfferingType, *serviceQuery)
+	if err != nil {
+		return nil, err
+	}
+	offerings := objectList.(*types.ServiceOfferings)
+	if offerings.Len() < 1 {
+		return nil, nil
+	}
+	brokerIDs := make([]string, 0, offerings.Len())
+	for _, p := range offerings.ServiceOfferings {
+		brokerIDs = append(brokerIDs, p.BrokerID)
```

Comment:
Should we remove the duplicates? 2 offerings might have the same BrokerID. Maybe use a map and then collect the keys.